apps/wallet-mobile/src/features/Receive/common/multipleAddressesModal.tsx

```diff
@@ -0,0 +1,45 @@
+import {parseSafe, useAsyncStorage, useMutationWithInvalidations} from '@yoroi/common'
+import {UseMutationOptions, useQuery} from 'react-query'
+
+import {useSelectedWallet} from '../../../SelectedWallet'
+
+const queryKey = 'showMultipleAddressesModal'
+
+export const useWriteMultipleAddressesModal = ({...options}: UseMutationOptions<void, Error> = {}) => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const mutation = useMutationWithInvalidations({
+    mutationFn: () => storage.join(`wallet/${wallet.id}/`).setItem(queryKey, true),
+    invalidateQueries: [[queryKey]],
+    ...options,
+  })
+
+  return mutation.mutate
+}
+
+export const useReadMultipleAddressesModal = () => {
```

Comment:
useMultipleAddressesModal

Reply:
@stackchain  we have this one that is using read and write in storage as 
```
export const useMultipleAddresses = () => {
  const readModalInfo = useReadMultipleAddressesModal()
  const writeModalInfo = useWriteMultipleAddressesModal()

  return {
    hideMultipleAddressesModal: () => writeModalInfo(),
    modalInfo: readModalInfo,
  }
}
```

will rename this one from `useMultipleAddresses` to `useMultipleAddressesModal`